ts/Extensions/BasePopup.ts

```diff
@@ -0,0 +1,189 @@
+/* *
+ *
+ *  Base popup.
+ *
+ *  (c) 2009 - 2023 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import AST from '../Core/Renderer/HTML/AST.js';
+import U from '../Core/Utilities.js';
+const {
+    addEvent,
+    createElement
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+abstract class BasePopup {
+
+    /* *
+     *
+     *  Constructor
+     *
+     * */
+
+    public constructor(
+        parentDiv: HTMLElement,
+        iconsURL: string
+    ) {
+        this.iconsURL = iconsURL;
+
+        this.container = this.createPopupContainer(parentDiv);
+
+        this.closeButton = this.addCloseButton();
+    }
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public container: HTMLElement;
+    public closeButton: HTMLElement;
+    public formType?: string;
+    public iconsURL: string;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /**
+     * Create popup div container.
+     *
+     * @param {HTMLElement} parentDiv
+     * Parent div to attach popup.
+     *
+     * @param  {string} className
+     * Class name of the popup.
+     *
+     * @return {HTMLElement}
+     * Popup div.
+     */
+    private createPopupContainer(
+        parentDiv: HTMLElement,
+        className: string = 'highcharts-popup highcharts-no-tooltip'
+    ): HTMLElement {
+        return createElement(
+            'div',
+            { className },
+            void 0,
+            parentDiv
+        );
+    }
+
+    /**
+     * Create HTML element and attach click event to close popup.
+     *
+     * @return {HTMLElement}
+     * Close button.
+     */
+    private addCloseButton(): HTMLElement {
+        const popup = this,
+            iconsURL = this.iconsURL;
+
+        // Create close popup button.
+        const closeButton = createElement(
+            'div',
+            {
+                className: 'highcharts-popup-close'
```

Comment:
It would be nice to set `className` as an argument - similar to `createPopupContainer`